### ISSUE TITLE
[HOLD E/E #311912] [$500] Assignee does not show on Confirm task page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#25680]
HELD for https://github.com/Expensify/Expensify/issues/311912

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. click on FAB icon and select new chat
2. enter email and create a new chat
3. click on FAB icon and select assign a task
4. Enter title and click on Next button
5. click on assignee and search for user with whom you created new chat
6.select that user
## Expected Result: 
Assignee does not show on Confirm task page
## Actual Result: 
Assignee should show on Confirm task page

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.56-2
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/3c32ed9e-91e2-447c-be42-e48ef4e22251

https://github.com/Expensify/App/assets/43996225/e0d52b04-d0ba-4b5f-8624-615a19ef6956

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692595469205799

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0182f226dc34b9a020</li>
        <li>Upwork Job ID: 1694007582851416064</li>
        <li>Last Price Increase: 2023-08-22</li>
<li>Automatic offers: </li>
<ul>
<li>gadhiyamanan | Reporter | 26281429</li>
</ul></li>
    </ul>
</details>

**Comment by b4s36t4** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Assignee does not show on Confirm task page
### What is the root cause of that problem?
This happened after #23773 in which we've updated the way we fetch the user details to display.

Here https://github.com/Expensify/App/blob/d38c06cc71ce90f0a890ddd6a208d849d0b13be3/src/libs/actions/Task.js#L719-L721 we're checking in the `personalDetails`.

Although we've initiated a chat with the User A, but User B haven't responded which is resulting personalDetails for the User B as undefined.

And here https://github.com/Expensify/App/blob/d38c06cc71ce90f0a890ddd6a208d849d0b13be3/src/libs/actions/Task.js#L721 we're returning an empty details which is causing the issue not to display anything in the Confirm task page.

### What changes do you think we should make in order to solve the problem?
Since we can't do anything about the user details not available we can return with the details we have. 

We can do default details instead of returning empty details which is causing the issues.

```jsx
if (!details) {
    return {
        icons: [{name: assignee, type: CONST.ICON_TYPE_AVATAR, source: UserUtils.getAvatar('', assigneeAccountID)}],
        displayName: assignee,
        subtitle: assignee,
    };
}
```

> `assignee` is passed form `NewTaskPage` which is `props.task.assignee`

The PR which caused the issue (maybe not) is have an issue with showing title for user icons, so we can follow the same approach we're following here to get the icon.

https://github.com/Expensify/App/blob/c5fc369015af5311c4c1207203d7d09c0010744d/src/libs/OptionsListUtils.js#L836-L845


### What alternative solutions did you explore? (Optional)
NA

## Result

https://github.com/Expensify/App/assets/59088937/b116143f-c00f-440a-8120-23fa0d889314

**Comment by DylanDylann** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Assignee does not show on Confirm task page

### What is the root cause of that problem?
When assignong a task to an assignee, we will check if chatReport with this assignee is existed or not
https://github.com/Expensify/App/blob/999ed938befa8910293cf349a4222314d5a73ccd/src/libs/actions/Task.js#L654-L673

If it does not exist, we will create an optimistic chat report and add this assignee to PERSONAL_DETAILS_LIST (on ONYX)

Then when displaying the assignee field we will get information from PERSONAL_DETAILS_LIST based on assigneeAccountID

But in this case, because we init the chat with this user before. So the condition check !chatReport is false
https://github.com/Expensify/App/blob/999ed938befa8910293cf349a4222314d5a73ccd/src/libs/actions/Task.js#L654

and this user will be not added to PERSONAL_DETAILS_LIST. So It displayed empty when displaying the assignee field 

### What changes do you think we should make in order to solve the problem?
when assigning new assignee we also should check if chat report exist and this assignee is not in PERSONAL_DETAILS_LIST, we will add this user to PERSONAL_DETAILS_LIST like here
https://github.com/Expensify/App/blob/999ed938befa8910293cf349a4222314d5a73ccd/src/libs/actions/Task.js#L666-L672

### What alternative solutions did you explore? (Optional)
When create new chat with new user, we also add this user to PERSONAL_DETAILS_LIST like here
https://github.com/Expensify/App/blob/999ed938befa8910293cf349a4222314d5a73ccd/src/libs/actions/Task.js#L666-L672

**Comment by parteekcoder** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue
After selecting an assignee, it still shows `Assignee` on Confirm task page

###  What is the root cause of that problem?
Here in this Menu option, there is no check that if an assignee is already chosen then don't show the `Assignee` option

https://github.com/Expensify/App/blob/main/src/pages/tasks/NewTaskPage.js#L168-#L175

```
         <MenuItem
                            label={assignee.displayName ? props.translate('task.assignee') : ''}
                            title={assignee.displayName || ''}
                            description={assignee.displayName ? LocalePhoneNumber.formatPhoneNumber(assignee.subtitle) : props.translate('task.assignee')}
                            icon={assignee.icons}
                            onPress={() => Navigation.navigate(ROUTES.NEW_TASK_ASSIGNEE)}
                            shouldShowRightIcon
                        />
```

### What changes do you think we should make in order to solve the problem?
We can add a simple condition here that if we have already chosen assignee then don't show this assignee option
```
      !props.task.assignee &&   <MenuItem
                            label={assignee.displayName ? props.translate('task.assignee') : ''}
                            title={assignee.displayName || ''}
                            description={assignee.displayName ? LocalePhoneNumber.formatPhoneNumber(assignee.subtitle) : props.translate('task.assignee')}
                            icon={assignee.icons}
                            onPress={() => Navigation.navigate(ROUTES.NEW_TASK_ASSIGNEE)}
                            shouldShowRightIcon
                        />
```

### What alternative solutions did you explore? (Optional)

## Supporting videos
Before:
[Screencast from 23-08-23 01:17:32 AM IST.webm](https://github.com/Expensify/App/assets/100110395/1752e146-2f08-4cd7-94a2-6cd986455e10)

After:
[Screencast from 23-08-23 01:16:24 AM IST.webm](https://github.com/Expensify/App/assets/100110395/056b200e-ad37-44fd-affb-bbaa9fc26244)

**Comment by DylanDylann** [comment]
@rushatgabhane When clicking a new user in the search page, we only call navigateToAndOpenReport and don't have any logic to create optimistic personalDetails. Only when this user send a message, the personalDetail will be responsed from BE and added to ONYX.

**Comment by mountiny** [comment]
@DylanDylann Can you please make a PR, thanks for review @rushatgabhane

**Comment by DylanDylann** [comment]
@rushatgabhane @mountiny The PR is ready for review https://github.com/Expensify/App/pull/25756
I can't run the auto workflow of PR. Anyone help me run all workflows

<img width="1042" alt="Screenshot 2023-08-24 at 13 47 54" src="https://github.com/Expensify/App/assets/141406735/70e62d71-ef3d-4bb0-9fe5-f21ef17472c6">

**Comment by CortneyOfstad** [comment]
Bump @rushatgabhane @mountiny ^^^

**Comment by mountiny** [comment]
We are still discussing if this solution is actually desired, its lower priority with the deadlines next week

**Comment by mountiny** [comment]
I believe this is help on internal PR to make this change possible

**Comment by CortneyOfstad** [comment]
Not overdue

**Comment by CortneyOfstad** [comment]
Any update @mountiny? TIA!

**Comment by mountiny** [comment]
No, this is very low priority at the moment

**Comment by CortneyOfstad** [comment]
No worries — thank you!

**Comment by CortneyOfstad** [comment]
Not overdue 👍

**Comment by CortneyOfstad** [comment]
Heading OoO until Oct. 25, so reassigning BZ to keep an eye on this while I'm gone 👍

**Comment by conorpendergrast** [comment]
@CortneyOfstad Welcome back! Assigning back to you; no change

**Comment by CortneyOfstad** [comment]
Perfect — thanks @conorpendergrast!

**Comment by CortneyOfstad** [comment]
@mountiny Any updates — TIA!

**Comment by mountiny** [comment]
No not really this is very low priority cc @thienlnam would you actually be interested in taking this one over as you got more context in the Tasks area so the fix might be quicker for you

**Comment by thienlnam** [comment]
Didn't read the entire thread but do you have a summary of the problem? Something with the personal details not showing up for a new user?

**Comment by mountiny** [comment]
It seems to be basically unable ot select the assignee in the new task flow when you just created a new chat with someone. I am double checking if we are even fixing these now

**Comment by thienlnam** [comment]
Ah yeah, I've seen this problem before - I believe it lies in that when you create optimistic personal details for someone you don't actually store the email as part of it until they respond to you

So when you select this user in the new task flow, it generates a new optimistic personal details for this person since it doesn't link to the existing contact but then when the task is created it pulls the existing user.

Anyways, it's not that simple to fix and I'd say not worth our time right now so we should just close this - thoughts?

**Comment by mountiny** [comment]
I have discussed this with Jason and I am going to create a project to collect these and close it, we are not going to focus on the task fixes right now

**Comment by mountiny** [comment]
Added it to the project and we will come back to this once we are ready to refocus on this feature

**Comment by DylanDylann** [comment]
@mountiny @CortneyOfstad Are we eligible for compensation for our effort? Contributors are already assigned and started on the PR. It's just due to the decision changes and the solution becomes outdated.

Some other similar cases where compensation is made 100% thought PR is closed
https://github.com/Expensify/App/issues/27108#issuecomment-1771000123
https://github.com/Expensify/App/issues/26470#issuecomment-1775949241

**Comment by mountiny** [comment]
I am sorry but in this case given the circumstances I would err on a side of no payout but rather just first dibs for you once we get back to this. We are going to fix this eventually but closing only to keep plate clener, otherwise we could put this on hold. Its added to a project in Todo category. This would require backend changes and external only work was not sufficient so I think full compensation at this point would not be appropriate either
